Add enable dashboard procedure (#527)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_importing-dashboards.adoc

[id="importing-dashboards_{context}"]
= Enabling dashboards

[role="_abstract"]

> The Grafana Operator can import and manage dashboards by creating `GrafanaDashboard` objects. A set of default dashboards can be enabled by Service Telemetry Operator which creates the `GrafanaDashboard` objects to result in dashboards loaded into the Grafana instance.

Reviewer: ```suggestion
The Grafana Operator can import and manage dashboards by creating `GrafanaDashboard` objects. Service Telemetry Operator can enable a set of default dashboards that create the `GrafanaDashboard` objects that load dashboards into the Grafana instance.
```

> The following dashboards are loaded into Grafana when `graphing.grafana.dashboards.enabled` is set to `true`:

Reviewer: ```suggestion
Set the value of `graphing.grafana.dashboards.enabled` to `true` to load the following dashboards into Grafana :
```

* Infrastructure dashboard
* Cloud view dashboard
* Virtual machine view dashboard
* Memcached view dashboard

> Additional dashboards can be created and loaded into Grafana by using the GrafanaDashboard object. For more information about managing dashboards with Grafana Operator, see link:https://grafana.github.io/grafana-operator/docs/dashboards/[Dashboards] in the _Grafana Operator project documentation_.

Reviewer: ```suggestion
You can use the `GrafanaDashboard` object to create and load additional dashboards into Grafana. For more information about managing dashboards with Grafana Operator, see link:https://grafana.github.io/grafana-operator/docs/dashboards/[Dashboards] in the _Grafana Operator project documentation_.
```

.Prerequisites

> * You've enabled graphing in the `ServiceTelemetry` object. For more information, see xref:setting-up-grafana-to-host-the-dashboard_assembly-advanced-features[].

Reviewer: ```suggestion
* You enabled graphing in the `ServiceTelemetry` object. For more information about graphing, see xref:setting-up-grafana-to-host-the-dashboard_assembly-advanced-features[].
```

.Procedure

> . To enable the managed dashboards, create or modify the `ServiceTelemetry` object. Set `graphing.grafana.dashboards.enabled` to `true`:

Reviewer: could this be 2 distinct steps?

Author: Single step. I was following an existing wording pattern somewhere else in the guide. You either create a new STF object, or modify one you've already created (which is the more likely scenario).

+
[source,bash]
----
$ oc edit stf default

apiVersion: infra.watch/v1beta1
kind: ServiceTelemetry
...
spec:
  ...
  graphing:
    enabled: true
    grafana:
      dashboards:
        enabled: true
----

> . Verify the Grafana dashboards have been created. It will take some time before the dashboards have been created by Service Telemetry Operator:

Reviewer: ```suggestion
. Verify that the Grafana dashboards are created. The process of Service Telemetry Operator creating the dashboards might take some time.
```

+
[source,bash,options="nowrap"]
----
$ oc get grafanadashboards.grafana.integreatly.org

NAME                          NO MATCHING INSTANCES   LAST RESYNC   AGE
memcached-dashboard-1                                 38s           38s
rhos-cloud-dashboard-1                                39s           39s
rhos-dashboard-1                                      39s           39s
virtual-machine-dashboard-1                           37s           37s
----

. Retrieve the Grafana route address:
+
[source,bash,options="nowrap"]
----
$ oc get route default-grafana-route -ojsonpath='{.spec.host}'

default-grafana-route-service-telemetry.apps.infra.watch
----

. In a web browser, navigate to https://_<grafana_route_address>_. Replace _<grafana_route_address>_ with the value that you retrieved in the previous step.

. Log in with OpenShift credentials. For more information about logging in, see xref:accessing-uis-for-stf-components_assembly-installing-the-core-components-of-stf[].

> . To view the dashboard, click *Dashboards* and *Browse*. The managed dashboards will be available in the _service-telemetry_ folder.

Reviewer: ```suggestion
. To view the dashboard, click *Dashboards* and *Browse*. The managed dashboards are available in the _service-telemetry_ folder.
```
